x-pack/plugins/security_solution/server/lib/telemetry/receiver.ts

```diff
@@ -445,6 +457,86 @@ export class TelemetryReceiver implements ITelemetryReceiver {
     return this.esClient.search<TelemetryEvent>(query);
   }
 
+  public async fetchDiagnosticAlertsBatch(
+    executeFrom: string,
+    executeTo: string,
+    pitId: string,
+    searchAfterValue: SortResults | undefined
+  ) {
+    if (this.esClient === undefined || this.esClient === null) {
+      throw Error('elasticsearch client is unavailable: cannot retrieve diagnostic alerts');
+    }
+
+    let newPitId = pitId;
+    let fetchMore = true;
+    let searchAfter: SortResults | undefined = searchAfterValue;
+
+    const query: ESSearchRequest = {
+      query: {
+        range: {
+          'event.ingested': {
+            gte: executeFrom,
+            lt: executeTo,
+          },
+        },
+      },
+      track_total_hits: false,
+      sort: [
+        {
+          'event.ingested': {
+            order: 'desc' as const,
+          },
+        },
+      ],
+      pit: { id: pitId },
+      search_after: searchAfter,
+      size: telemetryConfiguration.telemetry_max_buffer_size,
+      expand_wildcards: ['open' as const, 'hidden' as const],
+    };
+
+    let response = null;
+    try {
+      response = await this.esClient.search(query);
+      const numOfHits = response?.hits.hits.length;
+
+      if (numOfHits > 0) {
+        const lastHit = response?.hits.hits[numOfHits - 1];
+        searchAfter = lastHit?.sort;
+      }
+
+      fetchMore = numOfHits > 0 && numOfHits < 1_000;
```

Comment:
I don't get the `numOfHits < 1_000` part:

1. why the function returns false when there are more than 1K records?
2. in any case, shouldn't you use `telemetryConfiguration.telemetry_max_buffer_size`? which is the value used in the query `size` attribute?

Reply:
> why the function returns false when there are more than 1K records?

The ES client will return less than the specified amount when it has reached the last page (page likely being wrong terminology). This is the exit clause I have put into place. 

> in any case, shouldn't you use telemetryConfiguration.telemetry_max_buffer_size? which is the value used in the query size attribute?

Great catch!